### DISCUSSION TITLE
Why validation_step starts before training epoch ends?

### DISCUSSION THREAD

**Discussion by MooreManor** [Lightning-AI/pytorch-lightning#14269] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
As the pic is shown below, `validation_step` started before training epoch 0 ended.

![image](https://user-images.githubusercontent.com/49632327/185279513-7f376958-79aa-45e4-b5ea-940623d9d88e.png)

It seems that the validation progress bar rolls forward together with the training progress bar. How could I separate them (i.e. the training epoch first ends, and then validation starts) ?

**Comment by tshu-w** [comment, marked as answer]
Hi, the global process bar contains training + validation, this doesn't mean validation_step started before training epoch 0 ended.
https://github.com/Lightning-AI/lightning/issues/12623

**Comment by MooreManor** [comment]
@tshu-w 
Thanks for your clear explanation!

**Comment by MooreManor** [comment]
@tshu-w 
Does pytorch-lightning offer an API to change global progress bar to separated training and validation progress bar?

**Reply by tshu-w** [reply]
Not yet, see this issue https://github.com/Lightning-AI/lightning/issues/12623. BTW, users can write their own ProcessBar Callback.
